services/modelAdapter.js

```diff
@@ -0,0 +1,40 @@
+const config = require("../config.json");
+const {createChatCompletion} = require("./openAi");
+const {generateContent} = require("./vertexAi");
+const {generateClaudeContent} = require("./claudeAi");
+
+export default class ModelAdapter {
+    constructor() {
+        this.model_type = config.model_type;
```

Comment:
get the mode_type as a the ctor param - since the class has no meaning without it, and it should declare the initial params it needs to be existed